genie-core/src/main/java/com/netflix/genie/core/jpa/repositories/JpaJobRepository.java

```diff
@@ -84,4 +87,26 @@
      */
     // TODO: Explore deleteFirst{N}ByCreatedBefore
     Slice<IdProjection> findByCreatedBefore(@NotNull final Date date, @NotNull Pageable pageable);
+
+    /**
+     * Produce a count of jobs in the given states for each user.
+     *
+     * @param statuses the job statuses to filter by
+     * @return a set of {@link UserJobAmountAggregateProjection}
+     */
+    @Query("SELECT j.user, COUNT(j) FROM JobEntity j WHERE j.status IN (:statuses) GROUP BY j.user")
```

Comment:
Any particular reason to break this into two different queries and not just make one query that fetches both pieces of information count/memory?

Reply:
Didn't even think of this, would make this and downstream code simpler... How about going in with this for `3.3.x`, since it's close to EOL anyway, and do it as single query in `master`?

Reply:
Seems fine to me.